gpusim.cpp

```diff
@@ -40,6 +42,43 @@ const int DATABASE_VERSION = 3;
 
 namespace gpusim
 {
+
+class DecompressAssignFPRunnable : public QRunnable {
+public:
+    QByteArray compressed_data;
+    vector<char>& fp_vector;
+
+    DecompressAssignFPRunnable(vector<char>& fp_in) : fp_vector(fp_in) {};
+    void run() override {
+        QByteArray fp_qba = qUncompress(compressed_data);
+        compressed_data.clear();
+        fp_vector.reserve(fp_qba.size());
+        fp_vector.insert(fp_vector.begin(), fp_qba.data(), fp_qba.data()+fp_qba.size());
+        fp_qba.clear();
+    }
+};
+
+
+class DecompressAssignStringRunnable : public QRunnable {
+public:
+    QByteArray compressed_data;
+    vector<char*>& string_vector;
```

Comment:
why not std::string or QString?

Reply:
1) Qt deserializes strings as char*  2) Extra memory.  We can have billions of strings stored, the few extra bytes per string class matters.